thrifty-runtime/src/main/java/com/microsoft/thrifty/transport/FramedTransport.java

```diff
@@ -62,7 +62,9 @@ public int read(byte[] buffer, int offset, int count) throws IOException {
 
     private void readHeader() throws IOException {
         byte[] headerBytes = new byte[4];
-        inner.read(headerBytes, 0, headerBytes.length);
+        int numRead = inner.read(headerBytes, 0, headerBytes.length);
```

Comment:
As long as we're correcting this oversight, let's make sure we handle partial reads:

```
int numRead = 0;
while (numRead < headerBytes.length) {
  int n = inner.read(headerBytes, numRead, headerBytes.length - numRead);
  if (n == -1) {
    throw new EOFException();
  }
  numRead += n;
}
```

Reply:
Hi, thanks for review ) It's done! Maybe I wrong, but I think that two consecutive partial reads never happens because here data is reading in blocking way i.e. current thread just blocking until all needed data are read. If partial read is happen that mean no data left in stream and EOF is follow by.

Reply:
The thing here is that `Transport#read` makes no guarantee about that, so we have to assume that `read` can return before fully reading everything requested.  It's possible that `Buffer` makes that guarantee - I don't remember offhand - but certainly socket input streams can return without reading everything requested.

A partial read means EOF, certainly - but here it also means an incomplete or malformed data stream, possibly corrupted data.  Should we close un-exceptionally and just return `-1`?  I don't think it's clear that we should.  What do you think?

Reply:
Sorry for tardy answer. I think if we here read not enough data to parse header the best way is throw an exception because here is protocol violation, but -1 is some kind of 'normal' behavior.